### PR TITLE
test/system: Add test for journald log check in quadlet

### PR DESCRIPTION
**Pull request by ypu**
This test is used to ensure the default configuration for quadlets generated service files is sending stdout/stderr/syslog to the journald.
#### Does this PR introduce a user-facing change?

```NONE

```

### CODE REVIEW

File: test/system/252-quadlet.bats
Comment (Luap99) [inline test/system/252-quadlet.bats:1551]: This test takes over 10s because sleep does not react to SIGTERM. please use a different command or set StopTimeout=0 in the quadlet file

File: test/system/252-quadlet.bats
Comment (Luap99) [inline test/system/252-quadlet.bats:1546]: We already check the journal in `quadlet - oneshot`, I suggest you add you stdout/err check in this test for timing reasons

File: test/system/252-quadlet.bats
Comment (Luap99) [inline test/system/252-quadlet.bats:1564]: This is racy you first have to wait for the output to be logged but I really suggest to add it to the other test instead where it already works without flakes

**Review by Luap99 — CHANGES_REQUESTED** [review]

File: test/system/252-quadlet.bats
Comment (ypu) [reply test/system/252-quadlet.bats:1546]: Hi @Luap99 Thanks a lot for your advice. But in the oneshot case the point it write the logs to journald is a little different. In oneshot case the log is checked after the process is finished. But in this case want to check if we can get the logs during the process is still running just like a normal services. So I'd like to keep it a separated one.

File: test/system/252-quadlet.bats
Comment (ypu) [reply test/system/252-quadlet.bats:1551]: Got it. Will update the command.

File: test/system/252-quadlet.bats
Comment (ypu) [reply test/system/252-quadlet.bats:1564]: Understand it. Will add a wait for here to ignore that.

File: test/system/252-quadlet.bats
Comment (edsantiago) [reply test/system/252-quadlet.bats:1546]: Thank you for your PR. Can you clarify what the purpose of that testing would be? Is there some reason to believe that podman could possibly buffer journal output until a process terminates? There _is_ buffering / time delay going on, but [it is not podman's fault](https://github.com/systemd/systemd/issues/28650). And as @Luap99 mentioned, your test will flake due to this race condition.

File: test/system/252-quadlet.bats
Comment (edsantiago) [inline test/system/252-quadlet.bats:1557]: Due to the systemd race condition, this will have to be done in a while/timeout loop. (If you choose to continue with this testing approach).

File: test/system/252-quadlet.bats
Comment (edsantiago) [inline test/system/252-quadlet.bats:1562]: No undocumented magic constants, please, those make it very hard for future maintainers. This might be better:
```bash
    # log priority 3 == err
    run journalctl ... --priority 3 ...
```

**Review by edsantiago — CHANGES_REQUESTED** [review]
Thanks again for your PR. Please do clarify the perceived problem that this test addresses. A few comments inline should you wish to continue with this approach.

File: test/system/252-quadlet.bats
Comment (ypu) [reply test/system/252-quadlet.bats:1546]: This case want to check the services that started by the service file generated from quadlet can send stdout/stderr/syslog to the journal. Normally the services will keep running so from the user view there maybe have a slight different with the test case with oneshot. I am not saying there is problem exist in the podman write logs to journal during the process is running, but that is something that my test scenario want to covered and checked.

File: test/system/252-quadlet.bats
Comment (ypu) [reply test/system/252-quadlet.bats:1562]: Got it. Will update it. Thanks a lot for the advice.

File: test/system/252-quadlet.bats
Comment (Luap99) [reply test/system/252-quadlet.bats:1546]: It is fine to cover that but I think it still makes more sense to just extend an existing test for that.
The quadlet tests are not exactly fast as they need to generate/daemon-reload/start unit every time so I think it would help if you just add the print to an existing unit

**Comment by ypu** [comment]
Hi @edsantiago and @Luap99 Updated the tests based on your advices. Can you help to review it again? Thanks a lot!

File: test/system/252-quadlet.bats
Comment (Luap99) [reply test/system/252-quadlet.bats:1579]: I still prefer this being added to an existing test, i.e. `"quadlet - ContainerName`.
quadlet tests are slow (around 2.5s), it does not sounds like much but in the sum this adds up quickly.
As person doing a lot of work to speed system test up (https://github.com/containers/podman/pull/22826, https://github.com/containers/podman/pull/22821) we really need to be more careful when adding new tests.

File: test/system/252-quadlet.bats
Comment (edsantiago) [inline test/system/252-quadlet.bats:176]: Suggestion: `"Timed out waiting for '$expect_str' in journal"`

File: test/system/252-quadlet.bats
Comment (edsantiago) [inline test/system/252-quadlet.bats:275]: Suggestion: define `token_out="stdout$(random_string 10)"` and `token_err="stderr$(random_string 10)"` above, then use them in the echo and test.

File: test/system/252-quadlet.bats
Comment (edsantiago) [inline test/system/252-quadlet.bats:295]: Please use `!~` instead and remove the `.*`s

**Review by edsantiago — CHANGES_REQUESTED** [review]
Thank you for adding to the existing test. A few suggestions inline, and one blocker.

**Comment by ypu** [comment]
Hi @edsantiago Thanks a lot for your comments. Update the pullreq again. Can you help to review it? Thanks a lot!

**Comment by edsantiago** [comment]
Hi @ypu, I think we're stuck waiting for each other. Are you waiting for my review? I cannot review until CI passes, and CI is not passing. So I am waiting for you to fix that. I believe the current problem has been fixed in main, so it should be very easy to fix by simply rebasing.

At your convenience, please rebase and repush. Once CI passes, please feel free to ping me again.

**Review by edsantiago — APPROVED** [review]
LGTM, thank you!

**Review by Luap99 — APPROVED** [review]
/lgtm

**Pull request merged**
